modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/PartitionFilePageStoreFactory.java

```diff
@@ -0,0 +1,156 @@
+     * pageIdx == 0}) to create the {@link PartitionFilePageStore}.
+     *
+     * @param filePath File page store path.
+     * @param readIntoBuffer Direct byte buffer for reading {@link PartitionMetaIo meta} from {@code filePath}.
+     * @return File page store.
+     * @throws IgniteInternalCheckedException if failed
+     */
+    public PartitionFilePageStore createPageStore(
+            Path filePath,
+            ByteBuffer readIntoBuffer
+    ) throws IgniteInternalCheckedException {
+        assert readIntoBuffer.remaining() == pageSize : "Expected pageSize=" + pageSize + ", bufferSize=" + readIntoBuffer.remaining();
+
+        if (!Files.exists(filePath)) {
+            return createPageStore(filePath, new FilePageStoreHeader(latestVersion, pageSize), newPartitionMeta());
+        }
+
+        try (FileIo fileIo = fileIoFactory.create(filePath)) {
+            FilePageStoreHeader header = readHeader(fileIo);
+
+            if (header == null) {
+                header = new FilePageStoreHeader(latestVersion, pageSize);
+            }
+
+            PartitionMeta meta = readPartitionMeta(fileIo, header.headerSize(), readIntoBuffer);
+
+            return createPageStore(filePath, header, meta);
+        } catch (IOException e) {
+            throw new IgniteInternalCheckedException("Error while creating file page store [file=" + filePath + "]:", e);
+        }
+    }
+
+    private PartitionFilePageStore createPageStore(
+            Path filePath,
+            FilePageStoreHeader header,
+            PartitionMeta partitionMeta
+    ) throws IgniteInternalCheckedException {
+        if (header.version() == FilePageStore.VERSION_1) {
+            return new PartitionFilePageStore(
+                    new FilePageStore(
+                            header.version(),
+                            header.pageSize(),
+                            header.headerSize(),
+                            partitionMeta.pageCount(),
+                            filePath,
+                            fileIoFactory
+                    ),
+                    partitionMeta
+            );
+        }
+
+        throw new IgniteInternalCheckedException(String.format(
+                "Unknown version of file page store [version=%s, file=%s]",
+                header.version(),
+                filePath
+        ));
+    }
+
+    private PartitionMeta readPartitionMeta(FileIo fileIo, int headerSize, ByteBuffer readIntoBuffer) throws IOException {
+        if (fileIo.size() <= headerSize) {
+            return newPartitionMeta();
+        }
+
+        try {
+            fileIo.readFully(readIntoBuffer, headerSize);
+
+            long pageAddr = bufferAddress(readIntoBuffer);
+
+            if (PartitionMetaIo.getType(pageAddr) <= 0) {
+                return newPartitionMeta();
+            }
+
+            PartitionMetaIo partitionMetaIo = ioRegistry.resolve(pageAddr);
+
+            return new PartitionMeta(
```

Comment:
Can't we encapsulate this as well? I don't like that all methods of IO are explicitly listed somewhere outside of IO. Someone will forget to add line here while modifying Meta class, I promise you.